documentation/jetty-documentation/src/main/asciidoc/programming-guide/migration/migration-11-to-12.adoc

```diff
@@ -147,14 +153,32 @@ include::../{doc_code}/org/eclipse/jetty/docs/programming/migration/ServletToHan
 [[pg-migration-11-to-12-api-changes]]
 ==== APIs Changes
 
-===== `jetty-client`
+===== `HttpClient`
 
-`Request.onResponseContentDemanded(Response.DemandedContentListener)` has been replaced by `Request.onResponseContentSource(Response.ContentSourceListener)`.
+The Jetty 11 `Request.onResponseContentDemanded(Response.DemandedContentListener)` API has been replaced by `Request.onResponseContentSource(Response.ContentSourceListener)` in Jetty 12.
 
 However, also look at `Request.onResponseContentAsync(Response.AsyncContentListener)` and `Request.onResponseContent(Response.ContentListener)` for simpler usages.
 
-The old model was a "demand+push" model: the application was demanding content; when the content was available, the implementation was pushing content to the application by calling `DemandedContentListener.onContent(Response, LongConsumer, ByteBuffer, Callback)` for every content chunk.
+The Jetty 11 model was a "demand+push" model: the application was demanding content; when the content was available, the implementation was pushing content to the application by calling `DemandedContentListener.onContent(Response, LongConsumer, ByteBuffer, Callback)` for every content chunk.
 
-The new model is a "demand+pull" model: when the content is available, the implementation calls once `Response.ContentSourceListener.onContentSource(Content.Source)`; the application can then pull the content chunks from the `Content.Source`.
+The Jetty 12 model is a "demand+pull" model: when the content is available, the implementation calls once `Response.ContentSourceListener.onContentSource(Content.Source)`; the application can then pull the content chunks from the `Content.Source`.
 
 For more information about the new model, see xref:pg-arch-io-content-source[this section].
+
+===== WebSocket
+
+The Jetty WebSocket APIs have been vastly simplified.
+
+The Jetty 12 WebSocket APIs are now fully asynchronous, so the Jetty 11 `SuspendToken` class has been removed in favor of an explicit (or automatic) demand mechanism in Jetty 12 (for more information, refer to xref:pg-websocket-endpoints-demand[this section]).
+
+The various Jetty 11 `WebSocket*Listener` interfaces have been replaced by a single interface in Jetty 12, `Session.Listener.AutoDemanding` (for more information, refer to xref:pg-websocket-endpoints-listener[this section]).
+
+The Jetty 11 `RemoteEndpoint` APIs have been merged into the `Session` APIs in Jetty 12.
+
+The Jetty 11 `WriteCallback` class has been renamed to just `Callback` in Jetty 12, because it is now also used when receiving binary data.
```

Comment:
```suggestion
The Jetty 11 `WriteCallback` class has been renamed to just `Callback` in Jetty 12, because it is now also used when receiving binary data.  However, it is still a different interface to the `org.eclipse.jetty.util.Callback` interface, which cannot be used due to class loader visibility issues.
```